src/app/core/data/entity-type.service.ts

```diff
@@ -88,30 +88,10 @@ export class EntityTypeService extends DataService<ItemType> {
    * @param label
    */
   getEntityTypeByLabel(label: string): Observable<RemoteData<ItemType>> {
-
-    // TODO: Remove mock data once REST API supports this
-    /*
-    href$.pipe(take(1)).subscribe((href) => {
-      const request = new GetRequest(this.requestService.generateRequestId(), href);
-      this.requestService.configure(request);
+    let endPoint;
+    this.halService.getEndpoint(this.linkPath).pipe(take(1)).subscribe((result) => {
+      endPoint = result;
     });
-
-    return this.rdbService.buildSingle<EntityType>(href$);
-    */
-
-    // Mock:
-    const index = [
-      'Publication',
-      'Person',
-      'Project',
-      'OrgUnit',
-      'Journal',
-      'JournalVolume',
-      'JournalIssue',
-      'DataPackage',
-      'DataFile',
-    ].indexOf(label);
-
-    return this.findById((index + 1) + '');
+    return this.findByHref(endPoint + '/label/' + label);
```

Comment:
`endPoint` will only get a value if `this.halService.getEndpoint()` is synchronous. That won't be the case if it isn't already cached, or if the cache time runs out.

 What you want is something like:
```typescript
return this.halService.getEndpoint(this.linkPath).pipe(
  take(1),
  switchMap((endPoint: string) =>
    this.findByHref(endPoint + '/label/' + label))
);
```

That way `this.findByHref` will wait for the result of `this.halService.getEndpoint` even if it isn't synchronous

Reply:
That neat.  Didn't know I could return like that. Will Update and test this afternoon 9 ( EST ).

Reply:
Updated code and tested with local data: https://github.com/DSpace/dspace-angular/pull/965/commits/06bb2130783c2c042b1ea7698809fd3f0bec8124